File: program/program/find-sum-of-gp-series/FindSumOfGpSeries.cpp
//Programme to calculate the sum of Gp series.
// Formula --> (a(r^n - 1))/ (n -1)
#include <bits/stdc++.h>
#define ll long long
using namespace std;
ll solve(){

    return 0;
}
int main(){
    float a,r,n;  
    // here a--> first number of the Gp series.
    // r--> common ratio.
    // n --> the limit of the series.

    //Taking input;
    cout<<"Enter first number of the GP series : ";
    cin>>a;
    cout<<"Enter the common ratio of the GP series : ";
    cin>>r;
    cout<<"Enter the length of the series : ";
    cin>>n;
    
    // Calculations;
    float power = pow(r,n);
    float ans = (a*(power - 1)/(n-1)); 

    //Return ans;
    cout<<"The sum of the given GP series is: " <<ans<<endl;
    return 0;
}

/*
Contributed by : Kartikeya Saini
*/

File: program/program/find-sum-of-gp-series/FindSumOfGpSeries.fs
open System

let sumOfGP (a: float) (r: float) (n: int) : float =
    if r = 1.0 then float n * a
    else a * (pown r n - 1.0) / (r - 1.0)

// Function to safely parse float values
let tryParseFloat (str: string) : Option<float> =
    match Double.TryParse(str) with
    | (true, value) -> Some value
    | _ -> None

// Function to safely parse int values
let tryParseInt (str: string) : Option<int> =
    match Int32.TryParse(str) with
    | (true, value) -> Some value
    | _ -> None

// Read inputs from user
printfn "Enter the first term (a):"
let a = 
    match Console.ReadLine() |> tryParseFloat with
    | Some value -> value
    | None -> failwith "Invalid input for the first term"

printfn "Enter the common ratio (r):"
let r = 
    match Console.ReadLine() |> tryParseFloat with
    | Some value -> value
    | None -> failwith "Invalid input for the common ratio"

printfn "Enter the number of terms (n):"
let n = 
    match Console.ReadLine() |> tryParseInt with
    | Some value -> value
    | None -> failwith "Invalid input for the number of terms"

// Calculate and print the sum
printfn "Sum of GP series: %f" (sumOfGP a r n)